### DISCUSSION TITLE
Download instead of compiling?

### DISCUSSION THREAD

**Discussion by peterbe** [pyenv/pyenv#2900] [Ideas]
I love `pyenv` but I haven't been able to `pyenv install 3.11` or `pyenv install 3.12`. ...for ages. It's always something. [Here's how it looks today](https://gist.github.com/peterbe/cd36f854dccbaaa63f4d11facc32bc77).
I think it's been a year since this has been a problem. I try every now and then and tire at trying to understand the errors. I many times skimmed through [the wiki of common problems](https://github.com/pyenv/pyenv/wiki/Common-build-problems) to no avail. 

Surely, I've done something horrible on my macOS. When 3.10 came out, I think is when I installed `pyenv`. 
And clearly [I'm not alone](https://github.com/pyenv/pyenv/issues) :)

Anyway, I was watching https://rye-up.com/ https://www.youtube.com/watch?v=q99TYA7LnuA and saw that it uses ...

> Bootstraps Python: it provides an automated way to get access to the amazing [Indygreg Python Builds](https://github.com/indygreg/python-build-standalone/) as well as the PyPy binary distributions.

to get the binaries. I don't know much about it but I think the point is that it **downloads** the Pythons rather than *compiling* them. **Can we make `pyenv` do that too?** It could even become the default. Perhaps as an extension of `pyenv` it could become a new optional command:

```
# Downloads an already compiled version for my OS
> pyenv install --download 3.12

# Compiles a version for my OS
> pyenv install 3.12

```

**Comment by native-api** [comment]
We have an outstanding feature request for this: https://github.com/pyenv/pyenv/issues/2334

**Comment by native-api** [comment]
Duplicate of https://github.com/orgs/pyenv/discussions/1872
